frontend/my-adopthive-app/src/Profile/Profile.jsx

```diff
@@ -0,0 +1,35 @@
+import { useContext } from "react";
+import { UserContext } from "../UserContext";
+import { useNavigate } from "react-router-dom";
+import "./Profile.css";
+
+
+function Profile() {
+    const { user, updateUser } = useContext(UserContext);
+    const navigate = useNavigate();
+
+    function handleLogout() {
+        updateUser(null);
+        localStorage.removeItem("user");
+        navigate("/signin");
+    }
+
+    function setRole(role) {
+        if (role === "AP") {
+            return "Adoptive Parent";
+        } else if (role === "OS") {
+            return "Orphanage Staff";
+        }
```

Comment:
Might be cleaner if we process role with a switch-case statement instead of a sequence of if statements

Reply:
Noted. Thank you!